### PR TITLE
`--json` option for existing commands

### LINKED ISSUE #147: JSON output option for commands printing output
It'd be useful for IDE integration of Spring CLI to support JSON output option for commands listing projects, catalogs etc. Thus rather than receiving a difficult to parse ASCII table IDE could receive the easily convertible to POJOs JSON with the same data.

### PR DESCRIPTION
**Pull request by BoykoAlex**
Turns out that by returning an `Object` rather than `Table` or `String` command handler can support the `--json` option
Might fix #147 and replaces #148

### CODE REVIEW

**Comment by jvalkeal** [comment]
Yes returning Object has always been the not-so-obvious workaround.

File: src/main/java/org/springframework/cli/command/ProjectCommands.java [hunk 1]
```diff
@@ -80,62 +82,51 @@ public void projectAdd(
...
 
-		String[][] data = Stream.concat(header, allRows.stream()).toArray(String[][]::new);
+		Stream<String[]> header = Stream.<String[]>of(new String[] { "Name", "Description",  "URL", "Catalog", "Tags" });
+
+		// Retrieve project that were registered using the `project add` command and stored locally
+		Stream<String[]> rows = projectRepositories.stream()
+			.map(tr -> new String[] { tr.name(),
+					Objects.requireNonNullElse(tr.description, ""),
+					tr.url(),
+					"",
```
Comment (jvalkeal) [inline src/main/java/org/springframework/cli/command/ProjectCommands.java:125]: If this is left empty then `Catalog` column in table don't have content anymore. It used to come from `projectCatalog.getName()` in a code you refactored.

Comment (BoykoAlex) [reply src/main/java/org/springframework/cli/command/ProjectCommands.java]: Yes, i missed that - good catch. Should be fixed now.

**Comment by markpollack** [comment]
Great!  Merged as 8b5237423bc91e75600e41ee074a4ce3aab6ea08

**Pull request closed without merging**